ubireader/ubifs/walk.py

```diff
@@ -158,3 +160,25 @@ def index(ubifs, lnum, offset, inodes={}, bad_blocks=[]):
             inodes[ino_num]['dent']= []
 
         inodes[ino_num]['dent'].append(dn)
+
+    elif chdr.node_type == UBIFS_XENT_NODE:
+        try:
+            xn = nodes.xent_node(node_buf)
+
+        except Exception as e:
+            if settings.warn_only_block_read_errors:
+                error(index, 'Error', 'Problem at file address: %s extracting xent_node: %s' % (file_offset, e))
+                return
+            else:
+                error(index, 'Fatal', 'Problem at file address: %s extracting xent_node: %s' % (file_offset, e))
+        ino_num = xn.key['ino_num']
+        log(index, '%s file addr: %s, ino num: %s' % (xn, file_offset, ino_num))
+        verbose_display(xn)
+
+        if not ino_num in inodes:
+            inodes[ino_num] = {}
+
+        if not 'xent' in inodes[ino_num]:
+            inodes[ino_num]['xent']= []
+
+        inodes[ino_num]['xent'].append(xn)
```

Comment:
You could check here that `xn.name == UBIFS_XATTR_NAME_ENCRYPTION_CONTEXT` and lookup + decrypt the inode name. I suppose the extended nodes always follow their regular node counterpart.